eduHub/src/main/kotlin/com/arsiu/eduhub/config/bpp/NatsControllerBeanPostProcessor.kt

```diff
@@ -0,0 +1,26 @@
+package com.arsiu.eduhub.config.bpp
+
+import com.arsiu.eduhub.controller.nats.NatsController
+import com.google.protobuf.GeneratedMessageV3
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.stereotype.Component
+
+@Component
+class NatsControllerBeanPostProcessor : BeanPostProcessor {
+
+    override fun postProcessBeforeInitialization(bean: Any, beanName: String): Any {
```

Comment:
maybe its better to do this in postProcessAfterInitialization?

Reply:
When configuring callbacks for a NATS dispatcher, it's crucial to have them ready before your bean starts handling messages. To achieve this, you'd usually employ the postProcessBeforeInitialization phase to set up these callbacks. This ensures that your bean is fully primed and capable of processing messages right from the moment it's initialized.